src/chrome/chromeTargetDiscoveryStrategy.ts

```diff
@@ -55,7 +56,31 @@ export class ChromeTargetDiscovery implements ITargetDiscoveryStrategy, IObserva
         return this._getMatchingTargets(targets, targetFilter, targetUrl);
     }
 
+    private async _getVersionData(address: string, port: number): Promise<void> {
+
+        const url = `http://${address}:${port}/json/version`;
+        this.logger.log(`Getting browser and debug protocol version via ${url}`);
+
+        const jsonResponse = await utils.getURL(url, { headers: { Host: 'localhost' } })
+            .catch(e => utils.errP(localize('attach.cannotConnect', 'Cannot connect to the target: {0}', e.message)));
+
+        try {
+            const response = JSON.parse(jsonResponse);
+            this.logger.log(`Got browser version: ${response.Browser}`);
+            this.logger.log(`Got debug protocol version: ${response['Protocol-Version']}`);
+
+            this.telemetry.reportEvent('targetDebugProtocolVersion', { debugProtocolVersion: response['Protcol-Version'] });
+
+        } catch (e) {
+            return utils.errP(localize('attach.invalidResponse', 'Response from the target seems invalid. Error: {0}. Response: {1}', e.message, jsonResponse));
+        }
+    }
+
     private async _getTargets(address: string, port: number): Promise<ITarget[]> {
+
+        // Get the browser and the protocol version
+        await this._getVersionData(address, port);
```

Comment:
Can you have it only do this if /json/list fails? Or do you need this info all the time?

Reply:
What is your concern with doing this every time?

Reply:
I think this info will be helpful to us in all the cases irrespective of whether /list calls fails or not. Do you have any concerns?

Reply:
It's just doing extra work which will take a little time. How will it help when /list doesn't fail, you get the version info later too right?

Reply:
Yes but we don't get that info whenever there is a timeout which could be due to any other reasons as well and we do see a lot of customers having those.

Reply:
@roblourens What if we don't await this task, and we just do it in "background"? That way we are only spending CPU resources, and we are not making the user wait for this.

Reply:
> Yes but we don't get that info whenever there is a timeout which could be due to any other reasons as well and we do see a lot of customers having those.

A timeout for `/json/list`? But you would be getting it from /version when /json/list fails.

But yeah, it would be ok if you don't await it. And make sure it doesn't break anything if the runtime doesn't support /json/list/version. Right now it looks like it will fail if that endpoint doesn't exist.

Reply:
I made it not to await and handled the fail scenario to just log the error message.